backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -1446,4 +1450,127 @@ private boolean checkAllResourceAllowed(List<Resources> resources) {
+                              .setDateUpdated(Calendar.getInstance().getTimeInMillis())
+                              .setStartTime(Calendar.getInstance().getTimeInMillis())
+                              .setEndTime(Calendar.getInstance().getTimeInMillis());
+
+                          if (!request.getDestExperimentRunName().isEmpty()) {
+                            desExperimentRunBuilder.setName(request.getDestExperimentRunName());
+                          } else {
+                            desExperimentRunBuilder.setName(
+                                desExperimentRunBuilder.getName() + " - " + new Date().getTime());
+                          }
+
+                          if (userInfo != null) {
+                            desExperimentRunBuilder
+                                .clearOwner()
+                                .setOwner(userInfo.getVertaInfo().getUserId());
+                          }
+                          return createExperimentRunHandler
+                              .createClonedExperimentRun(desExperimentRunBuilder.build())
+                              .thenApply(
+                                  unused1 ->
+                                      CloneExperimentRun.Response.newBuilder()
+                                          .setRun(desExperimentRunBuilder.build())
+                                          .build(),
+                                  executor);
+                        },
+                        executor),
+            executor);
+  }
+
+  private InternalFuture<ExperimentRun.Builder>
+      checkPermissionAndPopulateFieldsBasedOnDestExperimentId(
+          CloneExperimentRun request, ExperimentRun.Builder cloneExperimentRunBuilder) {
+    if (!request.getDestExperimentId().isEmpty()) {
+      if (!cloneExperimentRunBuilder.getExperimentId().equals(request.getDestExperimentId())) {
+        return jdbi.withHandle(
+                handle ->
+                    handle
+                        .createQuery("SELECT project_id FROM experiment where id = :id")
+                        .bind("id", request.getDestExperimentId())
+                        .mapTo(String.class)
+                        .findOne())
+            .thenApply(
+                projectId -> {
+                  if (projectId.isEmpty()) {
+                    throw new NotFoundException(
+                        "Experiment not found for given ID: " + request.getDestExperimentId());
+                  }
+                  return projectId.get();
+                },
+                executor)
+            .thenCompose(
+                projectId -> {
+                  if (!projectId.equals(cloneExperimentRunBuilder.getProjectId())) {
+                    return getEntityPermissionBasedOnResourceTypes(
+                            Collections.singletonList(projectId),
+                            ModelDBActionEnum.ModelDBServiceActions.UPDATE,
+                            ModelDBServiceResourceTypes.PROJECT)
```

Comment:
We need to check permission even if the project IDs are equal. There's no guarantee that the user can update the project.

Reply:
Ok, i will update it

Reply:
Removed project id equal check